modules/tinymce/src/core/main/ts/DragDropOverrides.ts

```diff
@@ -121,6 +139,83 @@ const moveGhost = (
 
   ghostElm.style.width = (width - overflowX) + 'px';
   ghostElm.style.height = (height - overflowY) + 'px';
+
+  // Code needed for dragging CEF elements (specifically fixing TINY-8874)
+  // The idea behind the algorithm is that the user will start dragging the
+  // CEF element to the edge of the editor and that would cause scrolling.
+  // The way that happens is that the user will trigger a mousedown event,
+  // then a mousemove event until they reach the edge of the editor. Then
+  // no event triggers. That's when I set an interval to keep scrolling the editor.
+  // Once a new event triggers I clear the existing interval and set it back to none.
+
+  const clientHeight = contentAreaContainer.clientHeight;
+  const clientWidth = contentAreaContainer.clientWidth;
+
+  // current scroll level
+  const currentTop = win.scrollY;
+  const currentLeft = win.scrollX;
+
+  state.on((state) => {
+    if (state.dragging) {
+      // This basically means that the mouse is close to the bottom edge
+      // (within MouseRange pixels of the bottom edge)
+      if (mouseY + MouseRange >= clientHeight) {
+        const scrollDown = (currentTop: number) => {
+          win.scroll({
+            top: currentTop + ScrollingValue,
+            behavior: 'smooth'
+          });
+        };
+        scrollDown(currentTop);
+        state.intervalId = Optional.some(setInterval(() => {
+          const currentTop = win.scrollY;
+          scrollDown(currentTop);
+        }, IntervalValue));
+        // This basically means that the mouse is close to the top edge
+        // (within MouseRange pixels of the top edge)
+      } else if (mouseY - MouseRange <= 0) {
+        const scrollUp = (currentTop: number) => {
+          win.scroll({
+            top: currentTop - ScrollingValue,
+            behavior: 'smooth'
+          });
+        };
+        scrollUp(currentTop);
+        state.intervalId = Optional.some(setInterval(() => {
+          const currentTop = win.scrollY;
+          scrollUp(currentTop);
+        }, IntervalValue));
```

Comment:
For all of these, do you need to call the function before setting the interval? Does it really matter if nothing happens for the first 100ms?

If you remove that, the function can be inlined into the `setInterval` code.

Reply:
Done!